<launch>
  <arg name="test_mode" default="false" />

  <arg name="gui" default="true" unless="$(arg test_mode)" />
  <arg name="gui" default="false" if="$(arg test_mode)" />
  <arg name="gpu" default="0" unless="$(arg test_mode)"/>
  <arg name="gpu" default="-1" if="$(arg test_mode)" />

  <node name="image_publisher"
        pkg="jsk_perception" type="image_publisher.py">
    <rosparam subst_value="true">
      publish_info: false
      file_name: $(find jsk_perception)/sample/object_detection_example_1.jpg
    </rosparam>
  </node>
  <arg name="INPUT_IMAGE" default="image_publisher/output" />

  <node name="ssd_object_detector"
        pkg="jsk_perception" type="ssd_object_detector.py"
        output="screen">
    <remap from="~input" to="$(arg INPUT_IMAGE)" />
    <rosparam subst_value="true">
      gpu: $(arg gpu)
    </rosparam>
    <param name="model_path" value="voc0712" unless="$(arg test_mode)" />
  </node>

  <node name="image_view" pkg="image_view" type="image_view"
        if="$(arg gui)">
    <remap from="image" to="ssd_object_detector/output/image" />
  </node>
</launch>
